core/src/impl/Kokkos_Profiling_Interface.hpp

```diff
@@ -68,15 +76,49 @@ enum struct DeviceType {
   Unknown
 };
 
+struct ExecutionSpaceIdentifier {
+  DeviceType type;
+  uint32_t device_id;
+  uint32_t instance_id;
```

Comment:
`instance_id` here is theoretically 1:1 with, but not necessarily equal to, a CUDA/HIP stream ID, right? Is that mapping exposed somewhere?

Reply:
@wrwilliams, in the 99% case, you're correct that it's 1:1. A user _technically_ could create multiple distinct Kokkos::[Cuda/HIP] instances from the same stream. There's no valid reason to, but a person _could_.

That mapping is not exposed anywhere as of yet, I think a tool would need to discover it (something like monitor CUDA and see what stream a kernel was enqueued on). I could look into a scheme for this in a follow-on PR